core/src/main/resources/db/migration.sql

```diff
@@ -191,3 +191,83 @@ CREATE TABLE `structural_variant` (
+ADD UNIQUE INDEX `TMP_GENE_ID_UNIQUE` (`TMP_GENE_ID` ASC);
+
+-- populate genetic_entity
+insert into genetic_entity (entity_type, tmp_gene_id)
+(Select 'GENE', ENTREZ_GENE_ID from gene);
+
+-- update gene table to have GENETIC_ENTITY_ID point to the correct one:
+SET SQL_SAFE_UPDATES = 0;
+UPDATE gene
+INNER JOIN genetic_entity ON gene.ENTREZ_GENE_ID = genetic_entity.TMP_GENE_ID
+SET GENETIC_ENTITY_ID = genetic_entity.INTERNAL_ID;
+SET SQL_SAFE_UPDATES = 1;
+
+-- add UQ and FK constraint for GENETIC_ENTITY_ID in gene table:
+ALTER TABLE `gene` 
+CHANGE COLUMN `GENETIC_ENTITY_ID` `GENETIC_ENTITY_ID` INT NOT NULL,
+ADD UNIQUE INDEX `GENETIC_ENTITY_ID_UNIQUE` (`GENETIC_ENTITY_ID` ASC);
+
+ALTER TABLE `gene` 
+ADD CONSTRAINT `fk_gene_1`
+  FOREIGN KEY (`GENETIC_ENTITY_ID`)
+  REFERENCES `genetic_entity` (`INTERNAL_ID`)
+  ON DELETE NO ACTION
+  ON UPDATE NO ACTION;
+
+-- migrate genetic_alteration table in a similar way, pointing to GENETIC_ENTITY_ID 
+-- instead of ENTREZ_GENE_ID (note: the INSERT part can take some time [~20 min], 
+-- depending on how many studies you have in your DB):
+CREATE TABLE `genetic_alteration_new` (
+  `GENETIC_PROFILE_ID` int(11) NOT NULL,
+  `GENETIC_ENTITY_ID` int(11) NOT NULL,
+  `VALUES` longtext NOT NULL,
+  PRIMARY KEY (`GENETIC_PROFILE_ID`, `GENETIC_ENTITY_ID`),
+  CONSTRAINT `genetic_alteration_fk_1` FOREIGN KEY (`GENETIC_PROFILE_ID`) REFERENCES `genetic_profile` (`GENETIC_PROFILE_ID`) ON DELETE CASCADE,
+  CONSTRAINT `genetic_alteration_fk_2` FOREIGN KEY (`GENETIC_ENTITY_ID`) REFERENCES `genetic_entity` (`INTERNAL_ID`)
+);
+
+INSERT INTO genetic_alteration_new
+(GENETIC_PROFILE_ID, GENETIC_ENTITY_ID, `VALUES`)
+SELECT genetic_alteration.GENETIC_PROFILE_ID, genetic_entity.INTERNAL_ID, genetic_alteration.`VALUES`
+FROM genetic_alteration 
+INNER JOIN genetic_entity ON genetic_alteration.ENTREZ_GENE_ID = genetic_entity.TMP_GENE_ID;
+
+-- drop old genetic_alteration
+DROP TABLE genetic_alteration;
+-- rename new one to genetic_alteration:
+RENAME TABLE `genetic_alteration_new` TO `genetic_alteration`;
+-- drop temporary column:
+ALTER TABLE `genetic_entity` DROP COLUMN `TMP_GENE_ID`;
+
+DROP TRIGGER IF EXISTS `gene_BEFORE_DELETE`;
+-- BEFORE delete query to avoid the scenario where a gene is deleted and a genetic_entity is left "orphan"
+-- Definition needs to be on 1 row... (? test in python )
+CREATE DEFINER = CURRENT_USER TRIGGER `gene_BEFORE_DELETE` BEFORE DELETE ON `gene` FOR EACH ROW BEGIN  DELETE FROM genetic_entity where genetic_entity.INTERNAL_ID = OLD.GENETIC_ENTITY_ID; END;
+-- if you want to run above directly in DB, use 2 lines below:
+-- DELIMITER $$ 
+-- CREATE DEFINER = CURRENT_USER TRIGGER `gene_BEFORE_DELETE` BEFORE DELETE ON `gene` FOR EACH ROW BEGIN  DELETE FROM genetic_entity where genetic_entity.INTERNAL_ID = OLD.GENETIC_ENTITY_ID; END$$ DELIMITER ;
+
+
+
+-- ========================== end of genetic_entity related migration =============================================
```

Comment:
note .. comment skipping in migrate_db.sql depends on either applying #1807 or applying #1899